pkg/server/ca/manager/journal.go

```diff
@@ -233,3 +341,64 @@ func chainDER(chain []*x509.Certificate) [][]byte {
 	}
 	return der
 }
+
+// loadJournalFromDisk loads the journal from disk if it exists.
+// TODO: stop loading the journal from disk in v1.10 and remove this function.
+func loadJournalFromDisk(config *journalConfig) (*Journal, error) {
+	config.log.WithField(telemetry.Path, config.filePath).Debug("Loading journal from disk")
+
+	j := &Journal{
+		config:  config,
+		entries: new(journalEntries),
+	}
+
+	pemBytes, err := os.ReadFile(config.filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// There is no journal on disk. A new CA journal is created and will
+			// be stored in the next save operation.
+			return j, nil
+		}
+		return nil, errs.Wrap(err)
+	}
+	pemBlock, _ := pem.Decode(pemBytes)
+	if pemBlock == nil {
+		return nil, errs.New("invalid PEM block")
+	}
+	if pemBlock.Type != journalPEMType {
+		return nil, errs.New("invalid PEM block type %q", pemBlock.Type)
+	}
+
+	if err := proto.Unmarshal(pemBlock.Bytes, j.entries); err != nil {
+		return nil, errs.New("unable to unmarshal entries: %v", err)
+	}
+
+	return j, nil
+}
+
+// loadJournalFromDS loads the CA journal from the datastore.
+// It does that by looking for a CA journal record that matches with one of the
+// public keys of this server.
+func loadJournalFromDS(ctx context.Context, config *journalConfig) (*Journal, error) {
+	config.log.Debug("Loading journal from datastore")
+
+	j := &Journal{
+		config:  config,
+		entries: new(journalEntries),
+	}
+
+	caJournal, err := j.findCAJournal(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find CA journal record: %w", err)
+	}
+	if caJournal == nil {
+		j.config.log.Info("There is not a CA journal record that matches any of the local X509 authority IDs")
```

Comment:
I chose Info because not having a CA journal in the datastore seems to be a one-time condition to me: it's either because the journal hasn't been migrated from disk (during 1.9.x) yet or because it's being initialized.